File: python/Komputer.py
from Cpu import Cpu
from Harddrive import Harddrive
from Ram import Ram
from GPU import GPU
from Monitor import Monitor
from Motherboard import Motherboard
from PowerSupply import PowerSupply
from Casing import Casing

class Komputer:
    def __init__(self, nama="", cpu=None, ram_list=None, harddrive_list=None,
                 gpu=None, monitor=None, motherboard=None, power_supply=None, casing=None):
        self.nama = nama
        self.cpu = cpu if cpu is not None else Cpu()
        self.ram_list = ram_list if ram_list is not None else []
        self.harddrive_list = harddrive_list if harddrive_list is not None else []
        self.gpu = gpu if gpu is not None else GPU()
        self.monitor = monitor if monitor is not None else Monitor()
        self.motherboard = motherboard if motherboard is not None else Motherboard()
        self.power_supply = power_supply if power_supply is not None else PowerSupply()
        self.casing = casing if casing is not None else Casing()
    
    def set_nama(self, nama):
        self.nama = nama
    
    def set_cpu(self, cpu):
        self.cpu = cpu
    
    def set_ram_list(self, ram_list):
        self.ram_list = ram_list
    
    def set_harddrive_list(self, harddrive_list):
        self.harddrive_list = harddrive_list
    
    def set_gpu(self, gpu):
        self.gpu = gpu
    
    def set_monitor(self, monitor):
        self.monitor = monitor
    
    def set_motherboard(self, motherboard):
        self.motherboard = motherboard
    
    def set_power_supply(self, power_supply):
        self.power_supply = power_supply
    
    def set_casing(self, casing):
        self.casing = casing
    
    def add_ram(self, ram):
        self.ram_list.append(ram)
    
    def add_harddrive(self, harddrive):
        self.harddrive_list.append(harddrive)
    
    def get_nama(self):
        return self.nama
    
    def get_cpu(self):
        return self.cpu
    
    def get_ram_list(self):
        return self.ram_list
    
    def get_harddrive_list(self):
        return self.harddrive_list
    
    def get_gpu(self):
        return self.gpu
    
    def get_monitor(self):
        return self.monitor
    
    def get_motherboard(self):
        return self.motherboard
    
    def get_power_supply(self):
        return self.power_supply
    
    def get_casing(self):
        return self.casing
    
    def display_info(self):
        print("Informasi Komputer:")
        print(f"Nama      : {self.nama}")
        
        print("Cpu       : ", end="")
        self.cpu.display_info()
        print()
        
        print("Motherboard: ", end="")
        self.motherboard.display_info()
        print()
        
        for i, ram in enumerate(self.ram_list):
            print(f"Ram {i+1}      : ", end="")
            ram.display_info()
            print()
        
        for i, harddrive in enumerate(self.harddrive_list):
            print(f"Harddrive {i+1}: ", end="")
            harddrive.display_info()
            print()
        
        print("GPU       : ", end="")
        self.gpu.display_info()
        print()
        
        print("Power Supply: ", end="")
        self.power_supply.display_info()
        print()
        
        print("Casing    : ", end="")
        self.casing.display_info()
        print()
        
        print("Monitor   : ", end="")
        self.monitor.display_info()
        print()

File: cpp/Komputer.cpp
#ifndef KOMPUTER_CPP
#define KOMPUTER_CPP

#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"
#include "GPU.cpp"
#include "Monitor.cpp"
#include "Motherboard.cpp"
#include "PowerSupply.cpp"
#include "Casing.cpp"

using namespace std;

class Komputer {
private:
    string nama;
    Cpu cpu;
    vector<Ram> ramList;
    vector<Harddrive> harddriveList;
    GPU gpu;
    Monitor monitor;
    Motherboard motherboard;
    PowerSupply powerSupply;
    Casing casing;
public:
    Komputer() {}

    Komputer(string nama, Cpu cpu, vector<Ram> ramList, vector<Harddrive> harddriveList, 
             GPU gpu, Monitor monitor, Motherboard motherboard, PowerSupply powerSupply, Casing casing) {
        this->nama = nama;
        this->cpu = cpu;
        this->ramList = ramList;
        this->harddriveList = harddriveList;
        this->gpu = gpu;
        this->monitor = monitor;
        this->motherboard = motherboard;
        this->powerSupply = powerSupply;
        this->casing = casing;
    }

    void setNama(string nama) {
        this->nama = nama;
    }

    void setCpu(Cpu cpu) {
        this->cpu = cpu;
    }

    void setRamList(vector<Ram> ramList) {
        this->ramList = ramList;
    }    
   
    void setHarddriveList(vector<Harddrive> harddriveList) {
        this->harddriveList = harddriveList;
    }

    void setGPU(GPU gpu) {
        this->gpu = gpu;
    }

    void setMonitor(Monitor monitor) {
        this->monitor = monitor;
    }

    void setMotherboard(Motherboard motherboard) {
        this->motherboard = motherboard;
    }

    void setPowerSupply(PowerSupply powerSupply) {
        this->powerSupply = powerSupply;
    }

    void setCasing(Casing casing) {
        this->casing = casing;
    }

    void addRam(Ram ram) {
        this->ramList.push_back(ram);
    }

    void addHarddrive(Harddrive harddrive) {
        this->harddriveList.push_back(harddrive);
    }

    string getNama() {
        return this->nama;
    }

    Cpu getCpu() {
        return this->cpu;
    }

    vector<Ram> getRamList() {
        return this->ramList;
    }

    vector<Harddrive> getHarddriveList() {
        return this->harddriveList;
    }

    GPU getGPU() {
        return this->gpu;
    }

    Monitor getMonitor() {
        return this->monitor;
    }

    Motherboard getMotherboard() {
        return this->motherboard;
    }

    PowerSupply getPowerSupply() {
        return this->powerSupply;
    }

    Casing getCasing() {
        return this->casing;
    }

    void displayInfo() {
        cout << "Informasi Komputer:" << endl;
        cout << "Nama      : " << nama << endl;
        
        cout << "Cpu       : ";
        cpu.displayInfo();
        cout << endl;
        
        cout << "Motherboard: ";
        motherboard.displayInfo();
        cout << endl;
        
        for (int i = 0; i < ramList.size(); i++) {
            cout << "Ram " << (i+1) << "      : ";
            ramList[i].displayInfo();
            cout << endl;
        }
        
        for (int i = 0; i < harddriveList.size(); i++) {
            cout << "Harddrive " << (i+1) << ": ";
            harddriveList[i].displayInfo();
            cout << endl;
        }
        
        cout << "GPU       : ";
        gpu.displayInfo();
        cout << endl;
        
        cout << "Power Supply: ";
        powerSupply.displayInfo();
        cout << endl;
        
        cout << "Casing    : ";
        casing.displayInfo();
        cout << endl;
        
        cout << "Monitor   : ";
        monitor.displayInfo();
        cout << endl;
    }

    ~Komputer() {}
};

#endif // KOMPUTER_CPP
